VisualizationBase/src/overlays/IconOverlay.cpp

```diff
@@ -44,6 +44,9 @@ void IconOverlay::updateGeometry(int availableWidth, int availableHeight)
 {
 	Item::updateGeometry(icon_, availableWidth, availableHeight);
 
+	// match scale of the associated item
+	setScale(associatedItem()->totalScale());
```

Comment:
Please, add a bool property in the style and only set the scale if the property is true. We don't want to do this for all icons.